### PR TITLE
Move openlineage configuration to provider

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]

**Comment by potiuk** [comment]
Screenshot 2023-08-04 at 20 00 58

**Review by mobuchowski — APPROVED** [review]

File: airflow/providers/openlineage/plugins/openlineage.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 def _is_disabled() -> bool:
     return (
-        conf.getboolean("openlineage", "disabled")
+        conf.getboolean("openlineage", "disabled", fallback=False)
```
Comment (potiuk) [inline airflow/providers/openlineage/plugins/openlineage.py:28]: @mobuchowski - I had to add explicit fallback here as our "import" check checks the case where the providers are imported directly - and Provider's Manager might not yet have been initialized with provider configurations. So all top-level imports of configuration have to have explicit fallback, despite it being also specified in the provider.yaml

Comment (potiuk) [reply airflow/providers/openlineage/plugins/openlineage.py:28]: Our CI detected it BTW.

Comment (mobuchowski) [reply airflow/providers/openlineage/plugins/openlineage.py:28]: OK - makes sense. Top level means here "before initialization of plugins", right?

**Review by hussein-awala — APPROVED** [review]

File: airflow/providers/openlineage/plugins/openlineage.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/openlineage/plugins/openlineage.py:28]: yeah

Comment (potiuk) [reply airflow/providers/openlineage/plugins/openlineage.py:28]: Had to also add it for others.

**Pull request merged**